library/java/io/envoyproxy/envoymobile/engine/AndroidProxyMonitor.java

```diff
@@ -75,7 +78,12 @@ private ProxyInfo extractProxyInfo(final Intent intent) {
     // proxy is configured.
     //
     // See https://github.com/envoyproxy/envoy-mobile/issues/2531 for more details.
-    if (info.getPacFileUrl() != null && info.getPacFileUrl() != Uri.EMPTY) {
+    if (info.getHost() == "localhost" && info.getPort() == -1) {
```

Comment:
Is there a link to documentation or Android source code we could include in a comment here to explain why these magical values should mean there is no proxy?